sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1274,16 +1259,20 @@ class DataFrame private[sql](
    * @since 1.4.1
    */
   def drop(col: Column): DataFrame = {
```

Comment:
why we have this method....
we can only drop top level columns, allowing users to pass in a `Column` doesn't make sense.

cc @rxin @marmbrus